stratosphere/pm/source/impl/pm_resource_manager.cpp

```diff
@@ -0,0 +1,334 @@
+                    break;
+                }
+                svcSleepThread(1'000'000ul);
+            }
+        }
+
+    }
+
+    /* Resource API. */
+    Result InitializeResourceManager() {
+        /* Create resource limit handles. */
+        for (size_t i = 0; i < ResourceLimitGroup_Count; i++) {
+            if (i == ResourceLimitGroup_System) {
+                u64 value = 0;
+                R_ASSERT(svcGetInfo(&value, InfoType_ResourceLimit, INVALID_HANDLE, 0));
+                g_resource_limit_handles[i] = static_cast<Handle>(value);
+            } else {
+                R_ASSERT(svcCreateResourceLimit(&g_resource_limit_handles[i]));
+            }
+        }
+
+        /* Adjust resource limits based on firmware version. */
+        const auto firmware_version = GetRuntimeFirmwareVersion();
+        if (firmware_version >= FirmwareVersion_400) {
+            /* 4.0.0 increased the system thread limit. */
+            g_resource_limits[ResourceLimitGroup_System][LimitableResource_Threads] += ExtraSystemThreadCount400;
+            /* 4.0.0 also took memory away from applet and gave it to system, for the Standard and StandardForSystemDev profiles. */
+            g_memory_resource_limits[spl::MemoryArrangement_Standard][ResourceLimitGroup_System] += ExtraSystemMemorySize400;
+            g_memory_resource_limits[spl::MemoryArrangement_Standard][ResourceLimitGroup_Applet] -= ExtraSystemMemorySize400;
+            g_memory_resource_limits[spl::MemoryArrangement_StandardForSystemDev][ResourceLimitGroup_System] += ExtraSystemMemorySize400;
+            g_memory_resource_limits[spl::MemoryArrangement_StandardForSystemDev][ResourceLimitGroup_Applet] -= ExtraSystemMemorySize400;
+        }
+        if (firmware_version >= FirmwareVersion_500) {
+            /* 5.0.0 took more memory away from applet and gave it to system, for the Standard and StandardForSystemDev profiles. */
+            g_memory_resource_limits[spl::MemoryArrangement_Standard][ResourceLimitGroup_System] += ExtraSystemMemorySize500;
+            g_memory_resource_limits[spl::MemoryArrangement_Standard][ResourceLimitGroup_Applet] -= ExtraSystemMemorySize500;
+            g_memory_resource_limits[spl::MemoryArrangement_StandardForSystemDev][ResourceLimitGroup_System] += ExtraSystemMemorySize500;
+            g_memory_resource_limits[spl::MemoryArrangement_StandardForSystemDev][ResourceLimitGroup_Applet] -= ExtraSystemMemorySize500;
+        }
+        if (firmware_version >= FirmwareVersion_600) {
+            /* 6.0.0 increased the system event and session limits. */
+            g_resource_limits[ResourceLimitGroup_System][LimitableResource_Events]   += ExtraSystemEventCount600;
+            g_resource_limits[ResourceLimitGroup_System][LimitableResource_Sessions] += ExtraSystemSessionCount600;
+        }
+
+        /* 7.0.0+: Nintendo restricts the number of system threads here, from 0x260 -> 0x60. */
+        /* We will not do this. */
```

Comment:
♫┌( ͡° ͜ʖ ͡°)┘♪ ♫└( ͡° ͜ʖ ͡°)┐♪ 
In line with other places where we decided not to implement a Nintendo restriction (such as anti-dg checks), it might be worth leaving the corresponding code commented out for the sake of reference.

Reply:
Hm, good call.